### PR TITLE
add cross user network tests

### PR DESCRIPTION
**Pull request by mainred**
## High-level description

This PR introduces integration tests for communication between user networks, e.g. `calico` and `dcos`.

## Corresponding DC/OS tickets (required)

  - [D2IQ-63522](https://jira.d2iq.com/browse/D2IQ-63522) Integration test for checking the communication between mesos-overlay and calico networks.

## Related tickets (optional)

  - [D2IQ-ID](https://jira.mesosphere.com/browse/D2IQ-<number>) JIRA title / short description.

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain.
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by mainred** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by mainred** [comment]
@mesosphere-mergebot request-review  @jkoelker @wavesoft @vespian

File: packages/dcos-integration-test/extra/test_networking.py [hunk 1]
```diff
@@ -376,7 +387,70 @@ def test_calico_vip(dcos_api_session,
...
+@pytest.mark.slow
+@pytest.mark.parametrize(
+    'container,vip_network_name,proxy_network_name',
+    generate_vip_app_cross_usernetwork_permutations())
+def test_vip_cross_usernetwork(dcos_api_session,
+                               container: marathon.Container,
+                               vip_network_name: InternalUserNetwork,
+                               proxy_network_name: InternalUserNetwork,
+                               ipv6: bool=False):
+    '''Test the vip connective between different user networks
```
Comment (vespian) [inline packages/dcos-integration-test/extra/test_networking.py]:
```suggestion
    '''Test the vip connectivity between different user networks
```

Comment (mainred) [reply packages/dcos-integration-test/extra/test_networking.py]: Done, thanks :)

**Comment by jkoelker** [comment]
@mesosphere-mergebot changelog-not-required part of calico effort

**Comment by mainred** [comment]
@mesosphere-mergebot sync

**Comment by mainred** [comment]
@jkoelker @wavesoft @vespian, ping for another look.

**Pull request merged**